src/lib.rs

```diff
@@ -171,108 +170,146 @@ use std::sync::atomic::{AtomicIsize, Ordering, ATOMIC_ISIZE_INIT};
     pub heartbeat_frequency_ms: u32,
     /// Timeout for selecting an appropriate server for operations; default 30000 ms.
     pub server_selection_timeout_ms: i64,
     /// The size of the latency window for selecting suitable servers; default 15 ms.
     pub local_threshold_ms: i64,
-    /// Options for how to connect to the server.
-    pub stream_connector: StreamConnector,
 }
 
-impl ClientOptions {
-    /// Creates a new default options struct.
-    pub fn new() -> ClientOptions {
-        ClientOptions {
+impl Default for Connector {
+    fn default() -> Self {
+        Self {
             log_file: None,
-            read_preference: None,
-            write_concern: None,
             heartbeat_frequency_ms: DEFAULT_HEARTBEAT_FREQUENCY_MS,
             server_selection_timeout_ms: DEFAULT_SERVER_SELECTION_TIMEOUT_MS,
             local_threshold_ms: DEFAULT_LOCAL_THRESHOLD_MS,
-            stream_connector: StreamConnector::default(),
         }
     }
+}
+
+impl Connector {
+    /// Creates a new default options struct.
+    pub fn new() -> Self {
+        Default::default()
+    }
 
     /// Creates a new options struct with a specified log file.
-    pub fn with_log_file(file: &str) -> ClientOptions {
-        let mut options = ClientOptions::new();
-        options.log_file = Some(String::from(file));
-        options
+    pub fn log_file(&mut self, file: &str) -> &mut Self {
+        self.log_file = Some(file.to_string());
+        self
+    }
+
+    pub fn connect(&self, host: &str, port: u16) -> Result<Client> {
+        self.connect_with_connection_string(ConnectionString::new(host, port), Default::default())
+    }
+
+    pub fn connect_with_uri(&self, uri: &str) -> Result<Client> {
+        self.connect_with_connection_string(ConnectionString::parse(uri)?, Default::default())
     }
 
     #[cfg(feature = "ssl")]
-    /// Creates a new options struct with a specified SSL certificate and key files.
-    pub fn with_ssl(ca_file: &str,
-                    certificate_file: &str,
-                    key_file: &str,
-                    verify_peer: bool)
-        -> ClientOptions {
-            let mut options = ClientOptions::new();
-            options.stream_connector = StreamConnector::with_ssl(ca_file, certificate_file,
-                                                                 key_file, verify_peer);
-            options
+    pub fn connect_with_ssl(&self,
+                            uri: &str,
+                            ca_file: &str,
+                            certificate_file: &str,
+                            key_file: &str,
+                            verify_peer: bool)
+                            -> Result<Client> {
+        let connect_method =
+            ConnectMethod::with_ssl(ca_file, certificate_file, key_file, verify_peer);
+        self.connect_with_connection_string(ConnectionString::parse(uri)?, connect_method)
+    }
+
+
+    fn connect_with_connection_string(&self,
```

Comment:
This is essentially a copy of the old `Client::connect_with_config` except with the removal of read preference/write concern and the changing of variable names to reflect where they're now stored.